tests/checkers/unittest_utils.py

```diff
@@ -520,3 +520,32 @@ def test_is_typing_member() -> None:
     )
     assert not utils.is_typing_member(code[0], ("Literal",))
     assert not utils.is_typing_member(code[1], ("Literal",))
+
+
+def test_is_reassigned_after_current_requires_isinstance_check() -> None:
+    tree = astroid.parse(
+        """
+    CONSTANT = 1
+
+    def global_function_assign():
+        global CONSTANT
+        def CONSTANT():
+            pass
+        CONSTANT()
+    """
+    )
+    func = tree.body[1]
+    global_stmt = func.body[0]
+    nested_func = func.body[1]
+
+    assert isinstance(global_stmt, nodes.Global)
+    assert isinstance(nested_func, nodes.FunctionDef)
+
+    node_scope = global_stmt.scope()
+
+    assert nested_func.scope() == nested_func
```

Comment:
Thanks for the test, I see why you need the functiondef and class in the `isinstance` call now. At a glance, I'm surprised that a function's scope is itself. It's probably far too difficult to change now.